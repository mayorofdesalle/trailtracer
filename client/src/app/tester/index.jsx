import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
