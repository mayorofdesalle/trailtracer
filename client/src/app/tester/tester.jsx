import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };