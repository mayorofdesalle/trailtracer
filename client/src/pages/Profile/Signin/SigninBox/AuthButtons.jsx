import { useContext } from 'react';
import styled from 'styled-components';

import Button from '@components/ui/Button';
import Container from '@components/ui/Container';
import Icon from '@components/ui/Icon';
import PageContext from '@context/PageContext';

const AuthButtonsInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;

    & > ${Button} {
        --size: clamp(2.5rem, max(3dvw, 3dvh), 5rem);
        width: var(--size);
        height: var(--size);

        & > svg {
            height: 100%;
            width: 100%;
        }
    }
`;

/**
 * AuthButtons
 * @description
 * Styled container that wraps the auth buttons.
 **/
const AuthButtons = () => {
    const theme = useContext(PageContext).theme;
    const animColors = [theme.colors.background, theme.colors.secondary];

    return (
        <AuthButtonsInner>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='google-fill' color={theme.colors.primary} />
            </Button>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='facebook-fill' color={theme.colors.primary} />
            </Button>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='twitter-fill' color={theme.colors.primary} />
            </Button>
        </AuthButtonsInner>
    );
};

export default AuthButtons;