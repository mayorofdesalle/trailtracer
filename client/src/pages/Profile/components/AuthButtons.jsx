import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Button } from '@components/ui';
import { Container } from '@components/ui';
import { Icon } from '@components/ui';

const AuthButtonsInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;

    & > ${Button} {
        --size: clamp(2.5rem, max(3dvw, 3dvh), 5rem);
        width: var(--size);
        height: var(--size);

        & > svg {
            height: 100%;
            width: 100%;
        }
    }
`;

/**
 * AuthButtons
 * @description
 * Styled container that wraps the auth buttons.
 **/
const AuthButtons = ({ bgColor, color, animColors, ...props }) => {
    return (
        <AuthButtonsInner {...props}>
            <Button $bgColor={bgColor} $color={color} $animColors={animColors}>
                <Icon name='google-fill' />
            </Button>
            <Button $bgColor={bgColor} $color={color} $animColors={animColors}>
                <Icon name='facebook-fill' />
            </Button>
            <Button $bgColor={bgColor} $color={color} $animColors={animColors}>
                <Icon name='twitter-fill' />
            </Button>
        </AuthButtonsInner>
    );
};

AuthButtons.propTypes = {
    bgColor: PropTypes.string,
    color: PropTypes.string,
    animColors: PropTypes.array
};

export default AuthButtons;