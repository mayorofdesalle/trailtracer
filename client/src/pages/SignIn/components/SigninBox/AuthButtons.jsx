import styled, { useTheme } from 'styled-components';

import Container from '@components/ui/Container';
import Button from '@components/ui/Button';
import Icon from '@components/ui/Icon';

const AuthButtonsInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;
`;

const AuthButtons = () => {
    const theme = useTheme();
    const animColors = [theme.colors.background, theme.colors.secondary];

    return (
        <AuthButtonsInner>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='google-fill' color={theme.colors.primary} />
            </Button>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='facebook-fill' color={theme.colors.primary} />
            </Button>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='twitter-fill' color={theme.colors.primary} />
            </Button>
        </AuthButtonsInner>
    );
};

export default AuthButtons;