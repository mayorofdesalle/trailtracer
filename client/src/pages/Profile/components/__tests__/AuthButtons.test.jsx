import { describe, expect, it, render, screen } from '@app/tester';

import AuthButtons from '../AuthButtons';

describe('AuthButtons', () => {

    it('should render correctly', () => {
        render(<AuthButtons />);
        const buttons = screen.getAllByRole('button');
        expect(buttons).toHaveLength(3);
        buttons.forEach(button => expect(button).toBeInTheDocument());
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<AuthButtons />);
        expect(asFragment()).toMatchSnapshot();
    });

});