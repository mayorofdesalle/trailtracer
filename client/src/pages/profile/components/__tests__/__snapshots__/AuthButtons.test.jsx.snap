// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthButtons > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-kFCroH kHwwsn hqTKnQ"
  >
    <button
      class="sc-beySPh zbkUW"
      type="button"
    >
      <svg
        class="sc-eDLKkx fuXsHX"
        fill="currentColor"
        role="img"
      >
        <title>
          google-fill
        </title>
        <use
          href="http://localhost:3000/src/assets/icons.svg#google-fill"
        />
      </svg>
    </button>
    <button
      class="sc-beySPh zbkUW"
      type="button"
    >
      <svg
        class="sc-eDLKkx fuXsHX"
        fill="currentColor"
        role="img"
      >
        <title>
          facebook-fill
        </title>
        <use
          href="http://localhost:3000/src/assets/icons.svg#facebook-fill"
        />
      </svg>
    </button>
    <button
      class="sc-beySPh zbkUW"
      type="button"
    >
      <svg
        class="sc-eDLKkx fuXsHX"
        fill="currentColor"
        role="img"
      >
        <title>
          twitter-fill
        </title>
        <use
          href="http://localhost:3000/src/assets/icons.svg#twitter-fill"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
